### PR TITLE
James 4091 Ability to explore channels + Traffic shapping

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryWebAdminServerIntegrationTest.java [hunk 1]
```diff
@@ -39,4 +51,32 @@ class MemoryWebAdminServerIntegrationTest extends WebAdminServerIntegrationTest
...
+    TestIMAPClient testIMAPClient = new TestIMAPClient();
+
+    @Test
+    void shouldDescribeConnectedImapChannels(GuiceJamesServer server) throws Exception {
+        int imapPort = server.getProbe(ImapGuiceProbe.class).getImapPort();
+
+        server.getProbe(DataProbeImpl.class).addUser(USERNAME, PASSWORD);
+
+        testIMAPClient.connect(LOCALHOST_IP, imapPort);
+        testIMAPClient.sendCommand("ID (\"name\" \"Thunderbird\" \"version\" \"102.7.1\")");
```
Comment (HoussemNasri) [inline server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryWebAdminServerIntegrationTest.java:65]:
```suggestion
        testIMAPClient.sendCommand(""" ID ("name"  "Thunderbird"  "version"  "102.7.1") """);
```

Comment (chibenwa) [reply server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryWebAdminServerIntegrationTest.java:65]: image

TLDR choose your poinson" I'll keep the espace sequences.

**Review by quantranhong1999 — APPROVED** [review]

File: docs/modules/servers/partials/operate/webadmin.adoc [hunk 2]
```diff
@@ -4805,3 +4805,93 @@ Return code:
...
+....
+
+
+=== Listing all channels
+
+....
+curl -XGET /servers/channels
+....
+
+Will return a description and statistics for channels of a user:
```
Comment (Arsnael) [inline docs/modules/servers/partials/operate/webadmin.adoc:4867]:
```suggestion
Will return a description and statistics for channels of all users:
```

File: src/site/markdown/server/manage-webadmin.md [hunk 3]
```diff
@@ -4993,4 +4993,94 @@ Will disconnect `badGuy1@domain.tld` and `badGuy2@domain.tld`.
...
+```
+
+
+### Listing all channels
+
+```
+curl -XGET /servers/channels
+```
+
+Will return a description and statistics for channels of a user:
```
Comment (Arsnael) [inline src/site/markdown/server/manage-webadmin.md:5056]:
```suggestion
Will return a description and statistics for channels of all users:
```

File: src/site/markdown/server/manage-webadmin.md [hunk 4]
```diff
@@ -4993,4 +4993,94 @@ Will disconnect `badGuy1@domain.tld` and `badGuy2@domain.tld`.
...
 
 - 204: the certificate is reloaded
-- 400: Invalid request
\ No newline at end of file
+- 400: Invalid request
+
+### Listing connected users
+
+```
+curl -XDELETE /servers/connectedUsers
```
Comment (quantranhong1999) [inline src/site/markdown/server/manage-webadmin.md:5001]:
```suggestion
curl -XGET /servers/connectedUsers
```

File: docs/modules/servers/partials/operate/webadmin.adoc [hunk 5]
```diff
@@ -4805,3 +4805,93 @@ Return code:
 
 - 204: the certificate is reloaded
 - 400: Invalid request
+
+=== Listing connected users
+
+....
+curl -XDELETE /servers/connectedUsers
```
Comment (chibenwa) [inline docs/modules/servers/partials/operate/webadmin.adoc:4812]:
```suggestion
curl -XGET /servers/connectedUsers
```

**Comment by chibenwa** [comment]
Remaining work is to list connections for:

 - [ ] JMAP webSockets
 - [ ] JMAP event source

**Comment by Arsnael** [comment]
All the comments about doc suggestions have been marked as resolved, but I don't see any changes in the end in the final code change?

**Comment by chibenwa** [comment]
> All the comments about doc suggestions have been marked as resolved, but I don't see any changes in the end in the final code change?

Sorry I overwrote the web changes with local changes by mistake.

**Comment by Arsnael** [comment]
Conflict, please rebase

File: server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/TrafficShapingConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import io.netty.handler.traffic.ChannelTrafficShapingHandler;
+
+public record TrafficShapingConfiguration(long writeLimit, long readLimit, long checkInterval, long maxTime) {
+    static TrafficShapingConfiguration from(Configuration configuration) {
+        return new TrafficShapingConfiguration(
+            configuration.getLong("writeTrafficPerSecond", 0),
+            configuration.getLong("readTrafficPerSecond", 0),
+            configuration.getLong("checkInterval", 30),
+            configuration.getLong("maxDelays", 30));
```
Comment (HoussemNasri) [inline server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/TrafficShapingConfiguration.java:32]: Just looking at this part of the code it does not say what time unit we use for checkInterval and maxDelays, strong typing?

Comment (chibenwa) [reply server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/TrafficShapingConfiguration.java:32]: It mimics Netty configuration parameters,
I prefer to keep it alligned.

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]
